infra/resources/_modules/web_apps/pushnotif-func.tf

```
@@ -99,23 +139,47 @@ locals {
       # Notification Hubs variables
 
       # Endpoint for the test notification hub namespace
-      AZURE_NH_HUB_NAME                                = data.azurerm_notification_hub.common.name
+      AZURE_NH_HUB_NAME                                = data.azurerm_notification_hub.common_itn.name
       "AzureWebJobs.HandleNHNotificationCall.Disabled" = "0"
       # Endpoint for the test notification hub namespace
       NH1_PARTITION_REGEX = "^[0-3]"
-      NH1_NAME            = data.azurerm_notification_hub.common_partition[0].name
+      NH1_NAME            = data.azurerm_notification_hub.common_partition_itn[0].name
       NH2_PARTITION_REGEX = "^[4-7]"
-      NH2_NAME            = data.azurerm_notification_hub.common_partition[1].name
+      NH2_NAME            = data.azurerm_notification_hub.common_partition_itn[1].name
       NH3_PARTITION_REGEX = "^[8-b]"
-      NH3_NAME            = data.azurerm_notification_hub.common_partition[2].name
+      NH3_NAME            = data.azurerm_notification_hub.common_partition_itn[2].name
       NH4_PARTITION_REGEX = "^[c-f]"
-      NH4_NAME            = data.azurerm_notification_hub.common_partition[3].name
+      NH4_NAME            = data.azurerm_notification_hub.common_partition_itn[3].name
+
+      AZURE_NH_ENDPOINT = data.azurerm_key_vault_secret.azure_nh_endpoint_itn.value
+      NH1_ENDPOINT      = data.azurerm_key_vault_secret.azure_nh_partition1_endpoint_itn.value
+      NH2_ENDPOINT      = data.azurerm_key_vault_secret.azure_nh_partition2_endpoint_itn.value
+      NH3_ENDPOINT      = data.azurerm_key_vault_secret.azure_nh_partition3_endpoint_itn.value
+      NH4_ENDPOINT      = data.azurerm_key_vault_secret.azure_nh_partition4_endpoint_itn.value
```

We don't need to use a data to get those vaules, we can get them directly from the module.
I know that for the old ones we handled it like that but I prefer a different approach.
May I suggest you to take those values as input variables?
```suggestion
      NH1_ENDPOINT      = var.nh_partition_1.endpoint
      NH2_ENDPOINT      = var.nh_partition_2.endpoint
      NH3_ENDPOINT      = var.nh_partition_3.endpoint
      NH4_ENDPOINT      = var.nh_partition_4.endpoint
```

Then we can pass those values in `prod/web_apps.tf` with something like this
```tf
nh_partition_1 = module.notification_hubs_itn.partition_1
```
Probably we need to add an `output.tf` in `_modules/notification_hubs_itn`